diff --git a/ios/RNIContextMenu/RNIMenuActionItem.swift b/ios/RNIContextMenu/RNIMenuActionItem.swift
index 63fae48aeb169f0ec81798adc60f53304be9afcb..3e4ad9b2c5ba55108d7f447cb334a3f3c2221cde 100644
--- a/ios/RNIContextMenu/RNIMenuActionItem.swift
+++ b/ios/RNIContextMenu/RNIMenuActionItem.swift
@@ -182,7 +182,7 @@ extension RNIMenuActionItem {
     );
     
     #if swift(>=5.5)
-    if #available(iOS 15.0, *) {
+    if #available(iOS 16.0, *) {
       action.subtitle =
            self.actionSubtitle
         ?? self.fallbackActionSubtitle;
diff --git a/ios/RNIContextMenu/RNIMenuItem.swift b/ios/RNIContextMenu/RNIMenuItem.swift
index 6c14152eb1debcb258aa48f8602a1a7e09fcd320..f7c86ad4464e27a5591f309c390a0b1c1dc2062a 100644
--- a/ios/RNIContextMenu/RNIMenuItem.swift
+++ b/ios/RNIContextMenu/RNIMenuItem.swift
@@ -146,7 +146,7 @@ extension RNIMenuItem {
     );
     
     #if swift(>=5.5)
-    if #available(iOS 15.0, *),
+    if #available(iOS 16.0, *),
        let subtitle = self.menuSubtitle {
       
       menu.subtitle = subtitle;
